File: src/my_project/easy_problems/from151To200/reshape_matrix.py
from typing import List 

class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:

        if len(mat)*len(mat[0]) != r*c:
            return mat 

        solution_matrix = list()
        flat_list = list()

        for row in mat:
            for num in row:
                flat_list.append(num)
        k = 0
        for i in range(r):
            solution_matrix.append(flat_list[k:c+k])
            k+=c
                
        

        return solution_matrix

solution = Solution()

print(solution.matrixReshape( mat = [[1,2],[3,4]], r = 1, c = 4))

print(solution.matrixReshape(mat = [[1,2],[3,4]], r = 2, c = 4))

File: my_project_typescript/easy_problems_typescript/from51To100/reshapeMatrix.ts
function matrixReshape(mat: number[][], r: number, c: number): number[][] {

    if (mat.length*mat[0].length !== r*c){
        return mat
    }
    let solutionMatrix: Array<any> = []
    let flatList: Array<number> = []
    

    for (let row of mat){
        for (let num of row){
            flatList.push(num)
        }
    }
    
    let k: number = 0
    for (let i = 0; i <r; i++){
        solutionMatrix.push(flatList.slice(k, k+c))
        k += c
    }
    return solutionMatrix
};


console.log(matrixReshape([[1,2],[3,4]],  1, 4))

console.log(matrixReshape([[1,2],[3,4]], 2,  4))
